cli/resolver.rs

```diff
@@ -509,6 +509,14 @@ impl CliGraphResolver {
     self
   }
 
+  pub fn maybe_node_resolver(&self) -> Option<&CliNodeResolver> {
+    self.node_resolver.as_deref()
+  }
+
+  pub fn maybe_npm_resolver(&self) -> Option<&Arc<dyn CliNpmResolver>> {
+    self.npm_resolver.as_ref()
+  }
```

Comment:
Dependencies providing other dependencies (essentially acting like some kind of factory) is an indication that something is wrong in our code. Instead, we should look into other ways to improve the other code that needs these dependencies (ex. we could not use so many functions all around and instead use methods that don't take a dependency like this, but instead have a dependency is injected into it). Can you revert this change?

Reply:
@dsherret I added `LspResolver` which I think addresses this, since all the dependency getters are gone. A copy is available in the state snapshot.

I was going to add this eventually for multi-deno.json. It will encapsulate many scopes each as a `CliGraphResolver` and check the referrer's scope at resolution time. Right now it's root-only.